Packs/GitLab/Integrations/GitLabEventCollector/GitLabEventCollector.py

```diff
@@ -0,0 +1,129 @@
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from SiemApiModule import *  # noqa # pylint: disable=unused-wildcard-import
+
+
+class Client(IntegrationEventsClient):
+
+    def __init__(self, request: IntegrationHTTPRequest, options: IntegrationOptions, last_run: str):  # pragma: no cover
+        super().__init__(request=request, options=options)
+        self.last_run = last_run
+        self.page = 1
+        self.event_type = ''
+
+    def set_request_filter(self, after: Any):  # pragma: no cover
+        base_url = self.request.url.split("?")[0]
+        self.request.url = f'{base_url}?created_after={self.last_run[self.event_type]}&per_page=100&page={self.page}'
+        self.page += 1
+
+
+class GetEvents(IntegrationGetEvents):
+    @staticmethod
+    def get_last_run(events: list) -> dict:
+        groups = [event for event in events if event.get('entity_type') == 'Group']
+        groups.sort(key=lambda k: k.get('id'))
+        projects = [event for event in events if event.get('entity_type') == 'Project']
+        projects.sort(key=lambda k: k.get('id'))
+        user_events = [event for event in events if 'entity_type' not in event]
+        user_events.sort(key=lambda k: k.get('id'))
+        return {'groups': groups[-1]['created_at'], 'projects': projects[-1]['created_at'],
+                'events': user_events[-1]['created_at']}
+
+    def _iter_events(self):  # pragma: no cover
+        self.client.set_request_filter(None)
+        response = self.call()
+        events: list = response.json()
+        events.sort(key=lambda k: k.get('created_at'))
+        if not events:
+            return []
+
+        while True:
+            yield events
+            self.client.set_request_filter(None)
+            response = self.call()
+            events = response.json()
+            events.sort(key=lambda k: k.get('created_at'))
+            if not events:
+                break
+
+
+def main() -> None:  # pragma: no cover
+    demisto_params = demisto.params()
+    url = urljoin(demisto_params['url'], '/api/v4/')
+    events_collection_management = {
+        'groups_ids': argToList(demisto_params.get('group_ids', '')),
+        'projects_ids': argToList(demisto_params.get('project_ids', '')),
+        'event_types': ['groups', 'projects']
+    }
+    user_ids = demisto_params.get('user_ids', '').split(',')
```

Comment:
What are the `user_ids`?

Reply:
David send me the wrong url to the docs so I implemented according to it it was suppose to be all the events, I'll fix it tomorrow @yaakovi